kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerManagementIntegrationTest.java

```diff
@@ -343,6 +345,254 @@ public void testGetAndUpdateContainer() {
+        ContainerSpec containerToDeploy = new ContainerSpec(CONTAINER_ID, CONTAINER_NAME, serverTemplate, releaseId, KieContainerStatus.STOPPED, Collections.EMPTY_MAP);
+        try {
+            mgmtControllerClient.updateContainerSpec(serverTemplate.getId(), containerToDeploy);
+            fail("Container was created by update command - REST Post method.");
+        } catch (UnexpectedResponseCodeException e) {
+            assertEquals(404, e.getResponseCode());
+        }
+
+        KieServerAssert.assertNullOrEmpty("Found deployed container.", mgmtControllerClient.listContainerSpec(serverTemplate.getId()));
+    }
+
+    @Test
+    public void testUpdateContainerId() {
+        // Create kie server instance connection in controller.
+        ServerTemplate serverTemplate = createServerTemplate();
+
+        // Deploy container for kie server instance.
+        ContainerSpec containerToDeploy = new ContainerSpec(CONTAINER_ID, CONTAINER_NAME, serverTemplate, releaseId, KieContainerStatus.STOPPED, Collections.EMPTY_MAP);
+        mgmtControllerClient.saveContainerSpec(serverTemplate.getId(), containerToDeploy);
+
+        // Get container using kie controller.
+        ContainerSpec containerResponseEntity = mgmtControllerClient.getContainerInfo(kieServerInfo.getServerId(), CONTAINER_ID);
+        checkContainer(containerResponseEntity, KieContainerStatus.STOPPED);
+        assertEquals(CONTAINER_NAME, containerResponseEntity.getContainerName());
+
+        containerToDeploy.setId("newID");
+        try {
+            mgmtControllerClient.updateContainerSpec(serverTemplate.getId(), CONTAINER_ID, containerToDeploy);
+            fail("Container has updated id.");
+        } catch (UnexpectedResponseCodeException e) {
+            assertEquals(404, e.getResponseCode());
+            KieServerAssert.assertResultContainsString(e.getMessage(), "no container with id newID");
+        }
+    }
+
+    @Test
+    public void testUpdateContainerWithDifferrentID() {
+        ServerTemplate serverTemplate = createServerTemplate();
+        final String ONE_ID = "one";
+        final String TWO_ID = "two";
+
+        // Deploy container for kie server instance.
+        ContainerSpec containerOneToDeploy = new ContainerSpec(ONE_ID, ONE_ID, serverTemplate, releaseId, KieContainerStatus.STOPPED, Collections.EMPTY_MAP);
+        ContainerSpec containerTwoToDeploy = new ContainerSpec(TWO_ID, TWO_ID, serverTemplate, releaseId, KieContainerStatus.STOPPED, Collections.EMPTY_MAP);
+        mgmtControllerClient.saveContainerSpec(serverTemplate.getId(), containerOneToDeploy);
+        mgmtControllerClient.saveContainerSpec(serverTemplate.getId(), containerTwoToDeploy);
+
+        containerOneToDeploy.setReleasedId(releaseId101);
+
+        try {
+            mgmtControllerClient.updateContainerSpec(serverTemplate.getId(), TWO_ID, containerOneToDeploy);
+            fail("Container one was updated from container two REST endpoint.");
```

Comment:
This can be a reproducer for [JBPM-5931](https://issues.jboss.org/browse/JBPM-5931)